apps/webapp/src/state/ibc.ts

```diff
@@ -57,32 +62,37 @@ export const createIbcSendSlice = (): SliceCreator<IbcSendSlice> => (set, get) =
         state.ibc.destinationChainAddress = addr;
       });
     },
-    sendIbcWithdraw: async toastFn => {
+    sendIbcWithdraw: async () => {
       set(state => {
         state.send.txInProgress = true;
       });
 
-      const { dismiss, update } = toastFn(buildingTxToast());
+      const txToastId = toast(...buildingTxToast());
 
       try {
         const transactionPlan = await plan(await getPlanRequest(get().ibc));
-        const transaction = await authWitnessBuild(update, { transactionPlan });
-        const { txHash, detectionHeight } = await broadcast(update, { transaction });
+        const transaction = await authWitnessBuild({ transactionPlan }, status =>
+          toast(...buildingTxToast(status, txToastId)),
+        );
+        const txHash = await getTxHash(transaction);
+        const { detectionHeight } = await broadcast({ transaction }, status =>
+          toast(...broadcastingTxToast(txHash, status, txToastId)),
+        );
 
-        update(successTxToast(txHash, detectionHeight));
+        toast(...successTxToast(txHash, detectionHeight));
```

Comment:
`txHash` is available from `authWitnessBuild`, you don't have to calculate it independently

Reply:
this happens in the other slices as well

Reply:
@turbocrime it's available from `broadcast,` not `authWitnessBuild`, right? And since the `broadcastingTxToast` requires a `txHash`, I had to calculate the hash from the transaction, rather than getting it back from `broadcast`.